opml-importer.php

```diff
@@ -133,20 +144,26 @@ function dispatch() {
 			/** Load OPML Parser */
 			include_once( ABSPATH . 'wp-admin/link-parse-opml.php' );
 
-			$link_count = count($names);
+			$link_count    = count($names);
+			$link_inserted = 0;
 			for ( $i = 0; $i < $link_count; $i++ ) {
-				if ('Last' == substr($titles[$i], 0, 4))
-					$titles[$i] = '';
-				if ( 'http' == substr($titles[$i], 0, 4) )
-					$titles[$i] = '';
-				$link = array( 'link_url' => $urls[$i], 'link_name' => $wpdb->escape($names[$i]), 'link_category' => array($cat_id), 'link_description' => $wpdb->escape($descriptions[$i]), 'link_owner' => $user_ID, 'link_rss' => $feeds[$i]);
-				wp_insert_link($link);
-				echo sprintf('<p>'.__('Inserted <strong>%s</strong>', 'opml-importer').'</p>', $names[$i]);
+				$link = array( 'link_url' => $urls[$i], 'link_name' => esc_sql($names[$i]), 'link_category' => is_null($cat_id) ? null : array($cat_id), 'link_description' => esc_sql($descriptions[$i]), 'link_owner' => $user_ID, 'link_rss' => $feeds[$i]);
```

Comment:
Not sure `esc_sql()` is needed - [`wp_insert_link()`](https://developer.wordpress.org/reference/functions/wp_insert_link/) claims to sanitize all data it receives.